### PR TITLE
Tiny url improvements

### PR DESCRIPTION
**Pull request by wild-endeavor**
# TL;DR
This adds the ability to return an individual input or output in the flyte url, e.g.
```
flyte://v1/flytesnacks/development/ad5cb55f675g2s78swgp/end-node/i/o0
```

There's currently no mechanism by which these URLs can be returned to the frontend however.  This can be worked on separately.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
* Refactored out the parsing logic to return a union instead.  go doesn't have unions so returning a struct.
* Split out the GetData function into one for node level and one for task execution level.
* Added a second regex object to try to match against.

## Tracking Issue
NA

### CODE REVIEW

File: pkg/common/flyte_url_test.go [hunk 1]
```diff
@@ -172,3 +173,110 @@ func TestMatchRegexDirectly(t *testing.T) {
...
+	t.Run("regex with specific output no attempt", func(t *testing.T) {
+		matches := MatchRegex(reSpecificOutput, "flyte://v1/fs/dev/abc/n0/o/o0")
+		fmt.Println(matches)
+
+	})
+
+	t.Run("regex with specific output no attempt", func(t *testing.T) {
+		matches := MatchRegex(reSpecificOutput, "flyte://v1/fs/dev/abc/n0-dn0-9-n0-n0/o/o0")
+		fmt.Println(matches)
+	})
```
Comment (eapolinario) [inline pkg/common/flyte_url_test.go:187]: worth asserting in these tests?

**Review by eapolinario — DISMISSED** [review]
one minor comment, otherwise, LGTM.

File: pkg/common/flyte_url_test.go (hunk 1 above)
Comment (wild-endeavor) [reply pkg/common/flyte_url_test.go]: done

**Review by pingsutw — APPROVED** [review]

File: dataproxy/service.go [hunk 2]
```diff
@@ -347,11 +361,51 @@ func (s Service) GetData(ctx context.Context, req *service.GetDataRequest) (
...
+func (s Service) GetData(ctx context.Context, req *service.GetDataRequest) (
+	*service.GetDataResponse, error) {
+
+	logger.Debugf(ctx, "resolving flyte url query: %s", req.GetFlyteUrl())
+	err := s.validateResolveArtifactRequest(req)
+	if err != nil {
+		return nil, errors.NewFlyteAdminErrorf(codes.InvalidArgument, "failed to validate resolve artifact request. Error: %v", err)
+	}
+
+	executions, err := common.ParseFlyteURLToExecution(req.GetFlyteUrl())
```
Comment (eapolinario) [inline dataproxy/service.go:393]: nit: why is this in the plural?

**Review by eapolinario — DISMISSED** [review]
LGTM

**Review by eapolinario — APPROVED** [review]
